src/librustc_mir/interpret/eval_context.rs

```diff
@@ -1233,7 +1248,7 @@ impl<'a, 'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> EvalContext<'a, 'mir, 'tcx, M
         }
     }
 
-    pub fn read_value(&self, ptr: Scalar, align: Align, ty: Ty<'tcx>) -> EvalResult<'tcx, Value> {
+    pub fn read_value(&mut self, ptr: Scalar, align: Align, ty: Ty<'tcx>) -> EvalResult<'tcx, Value> {
```

Comment:
Why did this become `mut`?

Reply:
because the sanity check branch hadn't been merged yet and discriminant reading still allocated here

Reply:
So this can go back to just `&` now?